index.js

```diff
@@ -68,28 +112,14 @@ class ExecutorRouter extends Executor {
      * @return {Promise}
      */
     _start(config) {
-        const annotations = config.annotations || {};
-        const executorType = annotations[ANNOTATION_EXECUTOR_TYPE];
-        const executor = this[executorType] || this.defaultExecutor; // Route to executor (based on annotations) or use default executor
-
-        return executor.start(config);
-    }
+        const options = this.getFullConfig(config);
```

Comment:
Now that `.getFullConfig` can throw an error, you should either ensure that it's called within a Promise chain or catch it.